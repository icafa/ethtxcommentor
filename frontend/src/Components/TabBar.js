import React, { useState, useEffect } from 'react'
import { TabBar } from '@aragon/ui'
import history from '../history'

const AppTabBar = () => {
  const [selected, setSelected] = useState(0)

  function setCorrespondingTab(path) {
    if (path.includes('latesttransactions')) {
      setSelected(2)
    } else if (path.includes('transaction') || path.includes('transactions')) {
      setSelected(1)
    } else {
      setSelected(0)
    }
  }

  useEffect(() => {
    setCorrespondingTab(history.location.pathname)
  }, [])
  useEffect(() => {
    history.listen(location => {
      setCorrespondingTab(location.pathname)
    })
  }, [])
  function handleTabChange(index) {
    setSelected(index)
    switch (index) {
      case 0:
        history.push('/')
        break
      case 1:
        // replace this one with the latest block available
        history.push('/transactions/-1')
        break
      case 2:
        history.push('/latesttransactions')
        break;
      default:
        throw new Error(
          'Tab Index route change not implemented on the AppTabBar component.'
        )
    }
  }
  return (
    <TabBar
      items={['Blocks', 'Transactions', 'LatestTransactions']}
      selected={selected}
      onChange={handleTabChange}
    />
  )
}

export default AppTabBar
