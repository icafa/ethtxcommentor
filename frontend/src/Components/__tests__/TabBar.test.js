import React from 'react'
import { render } from '@testing-library/react'
import TabBar from '../TabBar'
import { Main } from '@aragon/ui'

describe('TabBarTest', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Main>
        <TabBar />
      </Main>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
