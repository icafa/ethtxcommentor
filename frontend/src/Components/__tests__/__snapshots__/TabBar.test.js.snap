// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarTest renders correctly 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <ul
        class="TabBar__Bar-sc-1rsszd9-0 euTWlF"
      >
        <li
          class="TabBar__Tab-sc-1rsszd9-2 kFRThM"
          tabindex="0"
        >
          <span
            class="TabBar__Label-sc-1rsszd9-3 gqxJuy"
          >
            Blocks
          </span>
        </li>
        <li
          class="TabBar__Tab-sc-1rsszd9-2 jPNMsb"
          tabindex="0"
        >
          <span
            class="TabBar__Label-sc-1rsszd9-3 ggJopK"
          >
            Transactions
          </span>
        </li>
      </ul>
    </nav>
    <div
      class="ToastHub__Container-sc-1y0i8xl-0 stXpK"
    />
  </div>
</DocumentFragment>
`;
